### PR TITLE
Lifecyclemanager

### PR DESCRIPTION
**Pull request by danielmarbach**
Another set of dependencies removed from the container.

### CODE REVIEW

**Comment by danielmarbach** [comment]
@Particular/azure-maintainers please review

File: src/Tests/Seam/When_dispatching_messages_in_receive_context.cs [hunk 1]
```diff
@@ -58,16 +51,29 @@ public void SetUp()
             endpointOrientedTopology.Initialize(settings);
 
             // create the topologySectionManager
-            var topologyCreator = (ICreateTopologyInternal) container.Resolve(typeof(TopologyCreator));
-            sectionManager = container.Resolve<ITopologySectionManagerInternal>();
-            await topologyCreator.Create(sectionManager.DetermineResourcesToCreate(new QueueBindings()));
-            container.RegisterSingleton<TopologyOperator>();
+            var namespaceManagerCreator = new NamespaceManagerCreator(settings);
+
+            var namespaceManagerLifeCycleManagerInternal = new NamespaceManagerLifeCycleManagerInternal(namespaceManagerCreator);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_dispatching_messages_in_receive_context.cs]: Probably nitpicking here, but still 
`namespaceManagerLifeCycleManagerInternal` is `Internal` really needed?

File: src/Tests/Seam/When_dispatching_messages_in_receive_context.cs [hunk 2]
```diff
@@ -58,16 +51,29 @@ public void SetUp()
...
+            var topologyCreator = new TopologyCreator(new AzureServiceBusSubscriptionCreatorV6(endpointOrientedTopology.Settings.SubscriptionSettings, settings), 
+                new AzureServiceBusQueueCreator(endpointOrientedTopology.Settings.QueueSettings,settings),  new AzureServiceBusTopicCreator(endpointOrientedTopology.Settings.TopicSettings), 
+                namespaceManagerLifeCycleManagerInternal);
+            var topologySectionManagerInternal = new EndpointOrientedTopologySectionManager(settings, container);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_dispatching_messages_in_receive_context.cs]: same question about `Internal`

File: src/Tests/Seam/When_dispatching_messages_in_receive_context.cs [hunk 3]
```diff
@@ -58,16 +51,29 @@ public void SetUp()
...
+            await topologyCreator.Create(topologySectionManagerInternal.DetermineResourcesToCreate(new QueueBindings()));
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_dispatching_messages_in_receive_context.cs]: `ConfigureAwait(flase)`?

File: src/Tests/Seam/When_dispatching_messages_in_receive_context.cs [hunk 4]
```diff
@@ -58,16 +51,29 @@ public void SetUp()
...
 
             // create the destination queue
-            var namespaceLifeCycle = new NamespaceManagerLifeCycleManagerInternal(new NamespaceManagerCreator(settings));
             var creator = new AzureServiceBusQueueCreator(endpointOrientedTopology.Settings.QueueSettings, settings);
-            namespaceManager = namespaceLifeCycle.Get("namespaceName");
+            namespaceManager = namespaceManagerLifeCycleManagerInternal.Get("namespaceName");
             await creator.Create(DestinationQueueName, namespaceManager);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_dispatching_messages_in_receive_context.cs]: `confgireAwait(false)` was missing before as well looks like

File: src/Tests/Seam/When_dispatching_messages_in_receive_context.cs [hunk 5]
```diff
@@ -58,16 +51,29 @@ public void SetUp()
...
+
+            dispatcher = endpointOrientedTopology.GetDispatcherFactory()();
+
+            timeToWaitBeforeTriggeringTheCircuitBreaker = TimeSpan.FromSeconds(5);
+            var createMessagingFactoriesInternal = new MessagingFactoryCreator(namespaceManagerLifeCycleManagerInternal, settings);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_dispatching_messages_in_receive_context.cs]: same question about `Internal`

File: src/Tests/Seam/When_dispatching_messages_in_receive_context.cs [hunk 6]
```diff
@@ -58,16 +51,29 @@ public void SetUp()
...
+            var manageMessagingFactoryLifeCycleInternal = new MessagingFactoryLifeCycleManager(createMessagingFactoriesInternal, settings);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_dispatching_messages_in_receive_context.cs]: same question about `Internal`

File: src/Tests/Seam/When_dispatching_messages_in_receive_context.cs [hunk 7]
```diff
@@ -58,16 +51,29 @@ public void SetUp()
...
+            var createMessageReceiversInternal = new MessageReceiverCreator(manageMessagingFactoryLifeCycleInternal, settings);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_dispatching_messages_in_receive_context.cs]: same question about `Internal`

File: src/Tests/Seam/When_dispatching_messages_in_receive_context.cs [hunk 8]
```diff
@@ -58,16 +51,29 @@ public void SetUp()
...
+            var clientEntities = new MessageReceiverLifeCycleManager(createMessageReceiversInternal, settings);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_dispatching_messages_in_receive_context.cs]: While this was a screwy name from the past, I'd still change it 😃

File: src/Tests/Seam/When_receiving_messages.cs [hunk 9]
```diff
@@ -29,10 +29,30 @@ public class When_receiving_messages
...
+            topology.Initialize(settings);
+
+            var extensions = new TransportExtensions<AzureServiceBusTransport>(settings);
+            settings.SetDefault("NServiceBus.Routing.EndpointName", "sales");
+            extensions.NamespacePartitioning().AddNamespace("namespaceName", AzureServiceBusConnectionString.Value);
+
+            // create the topologySectionManager
+            var namespaceManagerCreator = new NamespaceManagerCreator(settings);
+
+            var namespaceManagerLifeCycleManagerInternal = new NamespaceManagerLifeCycleManagerInternal(namespaceManagerCreator);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_receiving_messages.cs]: `Internal` in variable

File: src/Tests/Seam/When_receiving_messages.cs [hunk 10]
```diff
@@ -29,10 +29,30 @@ public class When_receiving_messages
...
+            var topologyCreator = new TopologyCreator(new AzureServiceBusSubscriptionCreatorV6(topology.Settings.SubscriptionSettings, settings), 
+                new AzureServiceBusQueueCreator(topology.Settings.QueueSettings,settings),  new AzureServiceBusTopicCreator(topology.Settings.TopicSettings), 
+                namespaceManagerLifeCycleManagerInternal);
+            var topologySectionManagerInternal = new EndpointOrientedTopologySectionManager(settings, container);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_receiving_messages.cs]: `Internal` in variable

File: src/Tests/Seam/When_receiving_messages.cs [hunk 11]
```diff
@@ -29,10 +29,30 @@ public class When_receiving_messages
...
+            await topologyCreator.Create(topologySectionManagerInternal.DetermineResourcesToCreate(new QueueBindings()));
 
             // setup the operator
-            var clientEntities = new MessageReceiverLifeCycleManager(new MessageReceiverCreator(new MessagingFactoryLifeCycleManager(new MessagingFactoryCreator(new NamespaceManagerLifeCycleManagerInternal(new NamespaceManagerCreator(settings)), settings), settings), settings), settings);
+            var createMessagingFactoriesInternal = new MessagingFactoryCreator(namespaceManagerLifeCycleManagerInternal, settings);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_receiving_messages.cs]: `Internal` in variable

File: src/Tests/Seam/When_receiving_messages.cs [hunk 12]
```diff
@@ -29,10 +29,30 @@ public class When_receiving_messages
...
+            var manageMessagingFactoryLifeCycleInternal = new MessagingFactoryLifeCycleManager(createMessagingFactoriesInternal, settings);
```
Comment (SeanFeldman) [inline src/Tests/Seam/When_receiving_messages.cs]: `Internal` in variable
I'll pause here as there's quite a few of them assuming you'll just rename the rest 😺

File: src/Transport/Connectivity/MessageReceiverLifeCycleManager.cs [hunk 13]
```diff
@@ -3,7 +3,7 @@ namespace NServiceBus.Transport.AzureServiceBus
     using System.Collections.Concurrent;
     using Settings;
 
-    class MessageReceiverLifeCycleManager : IManageMessageReceiverLifeCycleInternal
+    class MessageReceiverLifeCycleManager
```
Comment (SeanFeldman) [inline src/Transport/Connectivity/MessageReceiverLifeCycleManager.cs]: ❤️

File: src/Transport/Connectivity/MessageSenderLifeCycleManager.cs [hunk 14]
```diff
@@ -3,7 +3,7 @@ namespace NServiceBus.Transport.AzureServiceBus
     using System.Collections.Concurrent;
     using Settings;
 
-    class MessageSenderLifeCycleManager : IManageMessageSenderLifeCycleInternal
+    class MessageSenderLifeCycleManager
```
Comment (SeanFeldman) [inline src/Transport/Connectivity/MessageSenderLifeCycleManager.cs]: ❤️

File: src/Transport/Sending/DefaultOutgoingBatchRouter.cs [hunk 15]
```diff
@@ -14,15 +14,15 @@ class DefaultOutgoingBatchRouter : IRouteOutgoingBatchesInternal
     {
         ILog logger = LogManager.GetLogger<DefaultOutgoingBatchRouter>();
         IConvertOutgoingMessagesToBrokeredMessagesInternal outgoingMessageConverter;
-        IManageMessageSenderLifeCycleInternal senders;
+        MessageSenderLifeCycleManager senders;
```
Comment (SeanFeldman) [inline src/Transport/Sending/DefaultOutgoingBatchRouter.cs]: let's fix the name while we're at it?

File: src/Tests/Seam/When_dispatching_messages_in_receive_context.cs (hunk 3 above)
Comment (danielmarbach) [inline src/Tests/Seam/When_dispatching_messages_in_receive_context.cs]: In the core we do not enforce it in the tests. If you prefer to have it I'm happy

**Comment by danielmarbach** [comment]
Addressed comments

**Comment by SeanFeldman** [comment]
@Particular/azure-maintainers these changes cannot affect the failing tests as far as I can see.
What do we do?

**Comment by yvesgoeleven** [comment]
@Particular/azure-maintainers are these tests failing because of the subscription issue we investigated on friday?

**Comment by SeanFeldman** [comment]
@yvesgoeleven possibly.

**Comment by danielmarbach** [comment]
In the meantime this is green ;) Can we pull this in?

**Pull request merged**